environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -116,23 +203,114 @@ public Weld(String containerId) {
      *
      * @param extension an extension
      */
-    public void addExtension(Extension extension) {
+    public Weld addExtension(Extension extension) {
         if (extensions == null) {
             extensions = new HashSet<Metadata<Extension>>();
         }
         extensions.add(new MetadataImpl<Extension>(extension, "<explicitly-added>"));
+        return this;
+    }
+
+    /**
+     *
+     * @param interceptorClasses Interceptor classes enabled for a synthetic bean archive
+     * @return self
+     */
+    public Weld enableInterceptors(Class<?>... interceptorClasses) {
+        enabledInterceptors.clear();
+        for (Class<?> interceptorClass : interceptorClasses) {
+            enabledInterceptors.add(syntheticMetadata(interceptorClass));
+        }
+        return this;
+    }
+
+    /**
+     *
+     * @param decoratorClasses Decorator classes enabled for a synthetic bean archive
+     * @return self
+     */
+    public Weld enableDecorators(Class<?>... decoratorClasses) {
+        enabledDecorators.clear();
+        for (Class<?> decoratorClass : decoratorClasses) {
+            enabledDecorators.add(syntheticMetadata(decoratorClass));
+        }
+        return this;
+    }
+
+    /**
+     *
+     * @param alternativeClasses Alternative classes selected for a synthetic bean archive
+     * @return self
+     */
+    public Weld selectAlternatives(Class<?>... alternativeClasses) {
```

Comment:
I find the name of these enable/select methods quite verbose for a builder. Perhaps we could name them just interceptors(), decorators() and alternatives() and provide a more magical implementation that could enable the given artifact and add it to bean classes (both only if needed)?